pkg/operation/hybridbotanist/machines.go

```diff
@@ -171,7 +171,7 @@ func (b *HybridBotanist) DestroyMachines() error {
 
 	// Wait until all machine resources have been properly deleted.
 	if err := b.waitUntilMachineResourcesDeleted(machineClassPlural); err != nil {
-		return common.DetermineErrorCode(fmt.Sprintf("Failed while waiting for all machine resources to be deleted: '%s'", err.Error()))
+		return common.NewTerraformError(fmt.Sprintf("Failed while waiting for all machine resources to be deleted: '%s'", err.Error()))
```

Comment:
Let's not call the function `NewTerraformError` as it is also used here where Terraform does not play any role at all.